tests/testthat/test-merge_sce_list.R

```diff
@@ -578,17 +578,187 @@ test_that("merging SCEs where 1 altExp is missing works as expected, with altexp
+  sce2 <- sce_list_with_altexp[[2]]
+  other_altexp_name <- "other"
+  altExpNames(sce2) <- other_altexp_name
+  rownames(altExp(sce2)) <- c("ADT-A", "ADT-B", "ADT-C", "ADT-D", "ADT-E")
+
+  sce_list <- list(
+    "sce1" = sce1,
+    "sce2" = sce2
+  )
+
+  # Merge:
+  merged_sce <- merge_sce_list(
+    sce_list,
+    batch_column = batch_column,
+    # "total" should get removed
+    retain_coldata_cols = retain_coldata_cols,
+    # this row name should not be modified:
+    preserve_rowdata_cols = c("gene_names")
+  )
+
+  # Correct names
+  expect_equal(
+    altExpNames(merged_sce),
+    c(altexp_name, other_altexp_name)
+  )
+
+  # Check the "adt" (`altexp_name`) altexp
+  adt_merged <- altExp(merged_sce, altexp_name)
+  expect_equal(
+    dim(adt_merged),
+    c(num_altexp_features, ncol(merged_sce))
+  )
+  expect_equal(
+    rownames(adt_merged),
+    rownames(altExp(sce1))
+  )
+  expect_equal(
+    colnames(adt_merged),
+    colnames(merged_sce)
+  )
+  expect_true(
+    all(is.na(counts(adt_merged)[, merged_sce[[batch_column]] == "sce2"]))
+  )
+
+  # Check the "other"  altexp
+  other_merged <- altExp(merged_sce, "other")
+  expect_equal(
+    dim(other_merged),
+    c(num_altexp_features, ncol(merged_sce))
+  )
+  expect_equal(
+    rownames(other_merged),
+    rownames(altExp(sce2))
+  )
+  expect_equal(
+    colnames(adt_merged),
+    colnames(merged_sce)
+  )
+  expect_true(
+    all(is.na(counts(other_merged)[, merged_sce[[batch_column]] == "sce1"]))
+  )
+})
+
+
+
+test_that("merging SCEs with different altExps works as expected; each SCE has 2 different altExps", {
+  other_altexp_name <- "other"
+  other_n_features <- 3
+  sce1 <- sce_list_with_altexp[[1]]
+  sce2 <- sce_list_with_altexp[[2]]
+
+  sce_list <- list(
+    "sce1" = sce_list_with_altexp[[1]],
+    "sce2" = sce_list_with_altexp[[2]]
+  ) |>
+    # add "other" altExp to each sce
+    purrr::imap(
+      add_sce_altexp,
+      other_altexp_name,
+      other_n_features,
+      ncol(sce1)
+    )
+
+  # Merge
+  merged_sce <- merge_sce_list(
+    sce_list,
+    batch_column = batch_column,
+    # "total" should get removed
+    retain_coldata_cols = retain_coldata_cols,
+    # this row name should not be modified:
+    preserve_rowdata_cols = c("gene_names")
+  )
+
+  # Correct names
+  expect_equal(
+    altExpNames(merged_sce),
+    c(altexp_name, other_altexp_name)
+  )
+
+  # Check the "adt" (`altexp_name`) altexp
+  adt_merged <- altExp(merged_sce, altexp_name)
+  expect_equal(
+    dim(adt_merged),
+    c(num_altexp_features, ncol(merged_sce))
+  )
+  expect_equal(
+    rownames(adt_merged),
+    rownames(altExp(sce1, altexp_name))
+  )
+  expect_equal(
+    colnames(adt_merged),
+    colnames(merged_sce)
+  )
+
+  # next two tests check matrix values
+  expect_equal(
+    counts(adt_merged)[, merged_sce[[batch_column]] == "sce1"] |>
+      as.matrix() |>
+      unname(),
+    counts(altExp(sce_list[[1]], altexp_name)) |>
+      as.matrix() |>
+      unname()
```

Comment:
Since you have already check dimensions, we simplify these a bit by not worrying about whether we have a matrix for this stage:

```suggestion
    counts(adt_merged)[, merged_sce[[batch_column]] == "sce1"] |>
      as.numeric(),
    counts(altExp(sce_list[[1]], altexp_name)) |>
      as.numeric()
```

Reply:
Hopefully obviously, this applies everywhere you have an `as.matrix() |> unname()`